File: Factory Method/Dart/LoggerFactory.dart
import 'ErrorLogger.dart';
import 'ILogger.dart';
import 'InfoLogger.dart';
import 'LoggerType.dart';
import 'WarningLogger.dart';

class LoggerFactory {
  ILogger? create(LoggerType type) {
    switch (type) {
      case LoggerType.InfoLogger:
        return InfoLogger();
        break;

      case LoggerType.ErrorLogger:
        return ErrorLogger();
        break;

      case LoggerType.WarningLogger:
        return WarningLogger();
        break;

      default:
    }
  }
}

File: Factory Method/Python 3/LoggerFactory.py
from ErrorLogger import ErrorLogger
from InfoLogger import InfoLogger
from WarningLogger import WarningLogger
from LoggerType import LoggerType

class LoggerFactory:
  def create(self, type):
    if type == LoggerType.InfoLogger:
      return InfoLogger()
    elif type == LoggerType.WarningLogger:
      return WarningLogger()
    elif type == LoggerType.ErrorLogger:
      return ErrorLogger()
    
    return None